vignettes/data-combined.Rmd

```diff
@@ -0,0 +1,300 @@
+results.
+
+There are a few things to keep in mind here:
+
+- It doesn't matter in which order observed or simulated datasets are added.
+- You can add multiple `DataSet` objects in `$addDataSets()` method call.
+- You can add only a single instance of `SimulationResults` in `$addSimulationResults()` method call.
+- If you add a dataset with the same name as existing (e.g., adding simulation results 
+from multiple simulations and not specifying the name, so the path of the results 
+is used as the name), the new data will replace the existing.
+
+## Grouping
+
+Since the `DataCombined` object can store many datasets, some of them may naturally form a grouping and you would wish to track this in the object. Both `$addDataSets()` and `$addSimulationResults()` allow group specification via `groups` argument.
+
+When being plotted, data sets without a grouping will appear with distinct color, 
+line style (for simulated results), or symbol style (for observed data) with a 
+separate entry in the legend, with the name of the data set being the legend entry.
+
+Let's create a `DataCombined` and add simulation results and observed data *without* 
+specifying their grouping and create a time profile:
+
+```{r}
+myDataCombined <- DataCombined$new()
+
+myDataCombined$addSimulationResults(
+  simulationResults = simResults,
+  quantitiesOrPaths = outputPath,
+  names = "Aciclovir Plasma"
+)
+
+myDataCombined$addDataSets(
+  obsData$`Vergin 1995.Iv`
+)
+
+plotIndividualTimeProfile(dataCombined = myDataCombined)
+```
+
+If you do not specify `groups` when you add datasets or set it to `NULL`, and 
+wish to update groupings later, you can use the `$setGroups()` method. All data 
+within one group will get one entry in the legend with the name of the group, 
+and will be plotted with the same color.
+
+```{r}
+myDataCombined$setGroups(names = c("Aciclovir Plasma", obsData$`Vergin 1995.Iv`$name),
+                         groups = "Aciclovir PVB")
+
+plotIndividualTimeProfile(dataCombined = myDataCombined)
+```
+
+At any point, you can check the current names and groupings with the following active field:
+
+```{r}
+myDataCombined$groupMap
+```
+
+## Transformations
+
+Sometimes the raw data included in `DataCombined` needs to be transformed using specified
+offset and scale factor values.
+
+This is supported via `$setDataTransformations()` method, where you can specify the names of datasets, and offsets and scale factors. If these arguments are scalar (i.e., of length 1), then the same value will be applied to all specified names.
+
+The internal data frame in `DataCombined` will be transformed with the specified parameters, with the new columns computed as: 
+
+  - For x values: `newXValue = (rawXValue + xOffset) * xScaleFactor`
+  - For y values: `newYValue = (rawYValue + yOffset) * yScaleFactor`
+  - For arithmetic error: `newErrorValue = rawErrorValue * scaleFactor`
+  - For geometric errir: no transformation.
```

Comment:
I assume scale factors may be only >0?
At the moment, I can set negative scale factors without error:
```
> myDataCombined$setDataTransformations(forNames = "Aciclovir Plasma",xScaleFactors = -1)
> myDataCombined$setDataTransformations(forNames = "Aciclovir Plasma",yScaleFactors = -1)
```

interestingly, one of those -1 factors is reported as 1 after that:
```
> myDataCombined$dataTransformations
# A tibble: 3 x 5
  name               xOffsets yOffsets xScaleFactors yScaleFactors
  <chr>                 <dbl>    <dbl>         <dbl>         <dbl>
1 Aciclovir Plasma          0        0             1            -1
2 Blood Cells               0        0             1             1
3 Aciclovir Plasma 3        0        0             1             1
```

Reply:
> I assume scale factors may be only >0?

GitHub removed my long answer... Short version - I would not restrict it. E.g. inverting values by setting the scale factor to -1 to match measured transport rate with some PK-Sim transport rate that is negative... Very rare case, but why not.

> interestingly, one of those -1 factors is reported as 1 after that:

No, you just forgot to change the name in the second call, so you are setting a scale factor for the same data set twice :)

> I wonder if setting the Y-Offset != 0 for observed data with geometric error should result in an error?
Here the geoSD of transformed data cannot be calculated, so just "do nothing with geometric error" is in fact just wrong.

Created an issue for this.

Reply:
>GitHub removed my long answer... Short version - I would not restrict it. E.g. inverting values by setting the scale factor to -1 to match measured transport rate with some PK-Sim transport rate that is negative... Very rare case, but why not.

First side effect: setting the factor to negative we will become negative arithmetic error. 
But this is in fact an error in the formula: for the arithmetic error it should be then 
`newErrorValue = rawErrorValue * abs(scaleFactor)`

>No, you just forgot to change the name in the second call, so you are setting a scale factor for the same data set twice

but in the first call I set the **X**-Scalefactor and in the second call **Y**-ScaleFactor. Same dataset, yes, but 2 different factors

Reply:
> but in the first call I set the X-Scalefactor and in the second call Y-ScaleFactor. Same dataset, yes, but 2 different factors

Indeed! Will have to check it... next week.

Reply:
Now idea where it comes from, GitHub found the "extended" version of the answer....

> I assume scale factors may be only >0?

I don't see why we should restrict this to >0. I nevert thought about it, but theoretically the user can set all values to 0 by simply setting the scale factor to 0, or invert the values by setting the factor to -1 (as example, some measured rate, like a transport rate, that should be compared to a transport rate in PK-Sim that is defined in the "wrong" direction and is negative). Definitely not a common use case, but I would not forbid it.

> interestingly, one of those -1 factors is reported as 1 after that:

It is because in your code, you are setting the scale factor for the same name twice :)

```
> myDataCombined$setDataTransformations(forNames = **"Aciclovir Plasma"**,xScaleFactors = -1)
> myDataCombined$setDataTransformations(forNames = **"Aciclovir Plasma"**,yScaleFactors = -1)
```

Reply:
@Yuri05 This is the problem: https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/1120